### PR TITLE
BGDIINF_SB-2890 : add projection as props for all mapping components

### PR DESCRIPTION
**Pull request by pakb**
wherever component were not linked to the store yet. For others (those who already use the store) this will come later when the projection is in the store

### CODE REVIEW

File: src/modules/map/components/openlayers/OpenLayersGeoJSONLayer.vue [hunk 1]
```diff
@@ -46,62 +50,112 @@ export default {
...
+            if (!this.geojsonStyle) {
+                log.error('style was not loaded, could not create source')
+                return
+            }
+            // if the GeoJSON describes a CRS (projection) we grab it so that we can reproject on the fly if needed
+            const dataProjection = this.geojsonData.crs
+                ? this.geojsonData.crs.properties.name
+                : null
+
+            // reprojecting the GeoJSON if not in EPSG:3857, the default projection for GeoJSON is WGS84
```
Comment (ltshb) [inline src/modules/map/components/openlayers/OpenLayersGeoJSONLayer.vue:134]: Comment is not up to date

File: src/modules/map/components/openlayers/OpenLayersHighlightedFeature.vue [hunk 2]
```diff
@@ -59,6 +59,10 @@ export default {
             type: SelectableFeature,
             required: true,
         },
+        projection: {
+            type: CoordinateSystem,
+            default: LV95,
```
Comment (ltshb) [inline src/modules/map/components/openlayers/OpenLayersHighlightedFeature.vue:64]: Dito

File: src/modules/map/components/openlayers/OpenLayersInternalLayer.vue [hunk 3]
```diff
@@ -122,6 +128,14 @@ export default {
             type: Number,
             default: null,
         },
+        projection: {
+            type: CoordinateSystem,
+            default: LV95,
```
Comment (ltshb) [inline src/modules/map/components/openlayers/OpenLayersInternalLayer.vue:133]: Dito

File: src/modules/map/components/openlayers/OpenLayersKMLLayer.vue [hunk 4]
```diff
@@ -33,6 +33,10 @@ export default {
             type: Number,
             default: 1.0,
         },
+        projection: {
+            type: CoordinateSystem,
+            default: LV95,
```
Comment (ltshb) [inline src/modules/map/components/openlayers/OpenLayersKMLLayer.vue:38]: dito

File: src/modules/map/components/openlayers/OpenLayersWMSLayer.vue [hunk 5]
```diff
@@ -35,7 +34,7 @@ export default {
         },
         projection: {
             type: CoordinateSystem,
-            default: WEBMERCATOR.epsg,
+            default: LV95,
```
Comment (ltshb) [inline src/modules/map/components/openlayers/OpenLayersWMSLayer.vue:37]: Dito

**Review by ltshb — COMMENTED** [review]
I think we should have a config value for the default projection to be consistent

File: src/modules/drawing/components/DrawingMeasureInteraction.vue [hunk 6]
```diff
@@ -7,11 +7,18 @@ import { EditableFeatureTypes } from '@/api/features.api'
...
+import { CoordinateSystem, LV95 } from '@/utils/coordinateSystems'
 import { GeodesicGeometries } from '@/utils/geodesicManager'
 
 export default {
     mixins: [drawingInteractionMixin, drawingLineMixin],
     inject: ['getMap', 'getDrawingLayer'],
+    props: {
+        projection: {
+            type: CoordinateSystem,
+            default: LV95,
```
[the remark this answers is not in the document]
Comment (pakb) [reply src/modules/drawing/components/DrawingMeasureInteraction.vue]: I've added a config entry for this

**Review by ltshb — APPROVED** [review]
👍🏼

**Pull request merged**